src/coreclr/jit/lower.cpp

```diff
@@ -1523,6 +1523,77 @@ void Lowering::ReplaceArgWithPutArgOrBitcast(GenTree** argSlot, GenTree* putArgO
     BlockRange().InsertAfter(arg, putArgOrBitcast);
 }
 
+void Lowering::InsertSpecialCopyArg(GenTreePutArgStk* putArgStk, GenTreeLclVar* lclVar)
+{
+    assert(putArgStk != nullptr);
+    assert(lclVar != nullptr);
+
+    GenTree* dest;
+
+#if FEATURE_FIXED_OUT_ARGS
+    dest = comp->gtNewOperNode(GT_ADD, TYP_I_IMPL, comp->gtNewPhysRegNode(REG_SPBASE, TYP_I_IMPL),
+                               comp->gtNewIconNode(putArgStk->getArgOffset()));
+#else
+    dest = comp->gtNewPhysRegNode(REG_SPBASE, TYP_I_IMPL);
+#endif
```

Comment:
I think the `FEATURE_FIXED_OUT_ARGS` case is going to require more work than this. But also, for `FEATURE_FIXED_OUT_ARGS` we have an invariant that there are no calls between a `PUTARG_STK` and its corresponding `CALL` node (see `LegalizeArgPlacement`). How hard would it be to keep the non-x86 cases outside the JIT?

Reply:
Actually, we should be able to keep this invariant. I think the code just isn't working correctly for win-x64 as it is, since the destination here is not going to be the outgoing arg area (which this code computes), but rather the target of the implicit byref. So the code should instead be calling the copy constructor on same value as is being passed as the argument in the `FEATURE_FIXED_OUT_ARGS` case. I would add assert that the arg is `PassedByRef` as well.

Reply:
(Also, I would expect that a `PUTARG_REG` is possible in that case too.)